app/Http/Requests/Activity/Indicator/IndicatorRequest.php

```diff
@@ -123,33 +143,40 @@ protected function getMessagesForReference($formFields): array
      * @param $formFields
      *
      * @return array
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
      */
     protected function getRulesForBaseline($formFields): array
     {
         $rules = [];
 
         foreach ($formFields as $baselineIndex => $baseline) {
             $baselineForm = sprintf('baseline.%s', $baselineIndex);
+            $baselineYearRule = 'nullable|date_format:Y|digits:4';
 
-            $rules[sprintf('%s.year', $baselineForm)] = 'nullable|date_format:Y|digits:4';
+            if (!empty($baseline['date'])) {
+                $baselineYearRule = sprintf('%s|in:%s', $baselineYearRule, date('Y', strtotime($baseline['date'])));
+            }
+            $rules[sprintf('%s.year', $baselineForm)] = $baselineYearRule;
             $rules[sprintf('%s.value', $baselineForm)] = 'nullable|numeric|gte:0';
 
-            if ((request()->get('measure') == 2) &&
-                (Arr::get($baseline, 'value', null))) {
+            if ((request()->get('measure') == 2) && Arr::get($baseline, 'value', null)) {
```

Comment:
This if case might not be required since the similar rule is already mentioned on line 161